cmd/envbuilder/main.go

```diff
@@ -37,6 +37,7 @@ func envbuilderCmd() serpent.Command {
 		Options: o.CLI(),
 		Handler: func(inv *serpent.Invocation) error {
 			o.SetDefaults()
+			preExec := make([]func(), 0)
```

Comment:
Let's make sure we never miss a cleanup!
```suggestion
			var preExec []func()
			defer func() { // Ensure cleanup in case of error.
				for _, fn := range preExec {
					fn()
				}
			}()
```
Also small nit: Slightly cleaner to prefer var or `:= []func(){}` assignment over `make(..., 0)`.